tests/components/template/test_binary_sensor.py

```diff
@@ -671,3 +596,45 @@ async def test_unique_id(hass):
     await hass.async_block_till_done()
 
     assert len(hass.states.async_all()) == 1
+
+
+async def test_template_validation_error(hass, caplog):
+    """Test binary sensor template delay on."""
+    caplog.set_level(logging.ERROR)
+    config = {
+        "binary_sensor": {
+            "platform": "template",
+            "sensors": {
+                "test": {
+                    "friendly_name": "virtual thingy",
+                    "value_template": "True",
+                    "icon_template": "{{ states.sensor.test_state.state }}",
+                    "device_class": "motion",
+                    "delay_on": 5,
+                },
+            },
+        },
+    }
+    await setup.async_setup_component(hass, "binary_sensor", config)
+    await hass.async_block_till_done()
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    state = hass.states.get("binary_sensor.test")
+    assert state.attributes.get("icon") == ""
+
+    hass.states.async_set("sensor.test_state", "mdi:check")
+    await hass.async_block_till_done()
+
+    state = hass.states.get("binary_sensor.test")
+    assert state.attributes.get("icon") == "mdi:check"
+
+    hass.states.async_set("sensor.test_state", "invalid_icon")
+    await hass.async_block_till_done()
+    assert len(caplog.records) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)